Add a guide to Desktop Access Directory Sharing

ptgott opened the pull request:
Closes #15433

I placed this at the root of `docs/pages/desktop-access` since this
guide is currently the only step-by-step Desktop Access guide aside from
the Getting Started guide, and doesn't quite fit in the "Reference"
section.

Note that the "How it works" section at the bottom of the guide should
probably be in a separate architecture guide to Desktop Access. However,
we don't yet have a Desktop Access architecture guide, and I wanted to
include an architectural description here in case users want to
understand the architecture of the feature for threat modeling etc.

ptgott committed:
Add a guide to Desktop Access Directory Sharing

Closes #15433

I placed this at the root of `docs/pages/desktop-access` since this
guide is currently the only step-by-step Desktop Access guide aside from
the Getting Started guide, and doesn't quite fit in the "Reference"
section.

Note that the "How it works" section at the bottom of the guide should
probably be in a separate architecture guide to Desktop Access. However,
we don't yet have a Desktop Access architecture guide, and I wanted to
include an architectural description here in case users want to
understand the architecture of the feature for threat modeling etc.

zmb3 commented on docs/pages/desktop-access/directory-sharing.mdx:
```diff
@@ -0,0 +1,188 @@
+Once the session starts, click the three-dot menu on the top-right of the screen
+and click "Share Directory":
+
+![The "Share Desktop" button](../../img/desktop-access/share.png)
```
nit: should we clean up the desktop in the screenshot? there's a bunch of files and scripts scattered on screen which distracts from the key point (the share button)

zmb3 reviewed (approved):
Very well done, thanks Paul!

ptgott committed:
Respond to PR feedback

ibeckermayer commented on docs/pages/desktop-access/directory-sharing.mdx:
```diff
@@ -0,0 +1,188 @@
+
+After you grant permissions, you will see a new directory in File Explorer on
+the remote desktop. The directory will have the same name as the directory on
+your local machine, but with "on teleport" appended to the name. 
```
I think this is worthy of a screenshot below here of the File Explorer with the shared directory highlighted in the side bar.

<img width="1388" alt="image" src="https://user-images.githubusercontent.com/13578537/185966706-3ceb05b4-6671-4e26-872a-96214e8f9a01.png">

Reason being, I'm imagining that it might be slightly confusing for a user when they first share a directory, grant permissions, and then nothing much happens in the UI if they don't already have File Explorer open.

ibeckermayer commented on docs/pages/desktop-access/directory-sharing.mdx:
```diff
@@ -0,0 +1,188 @@
+the remote desktop. The directory will have the same name as the directory on
+your local machine, but with "on teleport" appended to the name. 
+
+You can only share a single directory at a time.
```
```suggestion
You can [currently](https://github.com/gravitational/teleport/issues/15715) only share a single directory at a time.
```

ptgott replied:
Is there a timeline for adding multi-directory support? Otherwise, I would rather have the docs represent the current state of the source.

ibeckermayer replied:
No, there's no timeline. The idea here was to make it easy to find the related issue so users can give it a 👍, in order to get a pulse on demand. Granted that the docs may not be an appropriate place for that, just a spontaneous idea.

ibeckermayer commented on docs/pages/desktop-access/directory-sharing.mdx:
```diff
@@ -0,0 +1,188 @@
+click "Disconnect." 
+
+The next time you start a session on the remote desktop, the directory will no
+longer be shared. You will need to share the directory again to access its
```
```suggestion
longer be shared. [Currently](https://github.com/gravitational/teleport/issues/15716), you will need to share the directory again to access its
```

ptgott replied:
Same as the previous suggestion

ibeckermayer commented on docs/pages/desktop-access/directory-sharing.mdx:
```diff
@@ -0,0 +1,188 @@
+Proxy Service, which forwards traffic to and from the relevant Teleport Desktop
+Service instance. 
+
+The Teleport Desktop Service converts traffic from the Teleport Web UI into RDP
```
```suggestion
The Teleport Desktop Service converts [TDP](https://github.com/gravitational/teleport/blob/master/rfd/0067-desktop-access-file-system-sharing.md) traffic from the Teleport Web UI into RDP
```

ptgott replied:
Added. I put a brief explanation of TDP before this paragraph and moved the link to a "You can read more..." paragraph at the end of this section to make the link target more explicit

ibeckermayer commented on docs/pages/desktop-access/directory-sharing.mdx:
```diff
@@ -0,0 +1,188 @@
+
+The Teleport Desktop Service converts traffic from the Teleport Web UI into RDP
+messages to send to the remote desktop. The Desktop Service also converts RDP
+traffic from the the remote desktop into messages to send to the Teleport Web
```
```suggestion
traffic from the the remote desktop into TDP messages to send to the Teleport Web
```

ibeckermayer commented on docs/pages/desktop-access/directory-sharing.mdx:
```diff
@@ -0,0 +1,188 @@
+  - auditor
++ - no-sharing
+  status:
+    is_locked:false
```
```suggestion
    is_locked: false
```

ibeckermayer commented on docs/pages/desktop-access/directory-sharing.mdx:
```diff
@@ -0,0 +1,188 @@
+longer be shared. You will need to share the directory again to access its
+content.
+
+## Step 4/4. Disable Directory Sharing
```
Nit: arguably a bit awkward to have this as a "step", but if this is what we do elsewhere then its not a big deal.

This is a helpful example, but I think we should also update the [RBAC](https://goteleport.com/docs/desktop-access/rbac/#clipboard-access) section of the docs (with that perhaps linking to here for a more detailed example).

ptgott replied:
I wanted to make this a step-by-step guide, and for me, putting this before we explain what Directory Sharing is wouldn't make as much sense. 

I'll plan to add a section to the RBAC page this week on this branch.

ptgott commented:
**TODO:** When we backport to v10, indicate on the v10 branch that this is in preview.

ptgott committed:
Address PR feedback

ptgott commented:
**TODO before we merge:**
- [ ] Add a section to the Desktop Access RBAC page re: disabling Directory Sharing
- [ ] Edit my branch to reflect the most recent changes to the feature

ibeckermayer commented on docs/pages/desktop-access/directory-sharing.mdx:
```diff
@@ -0,0 +1,204 @@
+about—or performs modifications on—the shared directory.
+
+You can read more about TDP in [Teleport RFD
+37](https://github.com/gravitational/teleport/blob/master/rfd/0067-desktop-access-file-system-sharing.md)
```
```suggestion
37](https://github.com/gravitational/teleport/blob/master/rfd/0037-desktop-access-protocol.md)
```

ptgott committed:
Respond to PR feedback and include updates

- Add an entry in the Desktop Access RBAC guide
- Fix a link
- Include updated notes on the product

ptgott commented:
Hi @ibeckermayer, when should this be mergeable? Thanks!
